pubsub/topics.js

```diff
@@ -128,75 +107,65 @@ function setPublishCounterValue (value) {
 }
 
 // [START pubsub_publish_ordered_message]
-function publishOrderedMessage (topicName, data, callback) {
+function publishOrderedMessage (topicName, data) {
   // Instantiates a client
-  const pubsubClient = PubSub();
+  const pubsub = PubSub();
 
   // References an existing topic, e.g. "my-topic"
-  const topic = pubsubClient.topic(topicName);
-
-  /**
-   * In Node.js, a PubSub message requires a "data" property, which can have a
-   * string or an object as its value. An optional "attributes" property can be
-   * an object of key/value pairs, where the keys and values are both strings.
-   * See https://cloud.google.com/pubsub/reference/rpc/google.pubsub.v1#google.pubsub.v1.PubsubMessage
-   *
-   * Topic#publish() takes either a single message object or an array of message
-   * objects. See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/topic?method=publish
-   */
+  const topic = pubsub.topic(topicName);
+
   const message = {
     data: data,
 
     // Pub/Sub messages are unordered, so assign an order id to the message to
     // manually order messages
     attributes: {
-      counterId: '' + getPublishCounterValue()
+      counterId: `${getPublishCounterValue()}`
     }
   };
 
-  topic.publish(message, (err, messageIds) => {
-    if (err) {
-      callback(err);
-      return;
-    }
+  // Publishes the message, use raw: true to pass a message with attributes
+  return topic.publish(message, { raw: true })
+    .then((results) => {
+      const messageIds = results[0];
+
+      // Update the counter value
+      setPublishCounterValue(+message.attributes.counterId + 1);
```

Comment:
obscure: maybe, I'll make it clearer.

unnecessary: it's definitely necessary, otherwise you'd get `"51"` instead of `6`.